File: Web/src/app/models/records/time-off.ts
import { Employee } from './employee';


export class TimeOff {
  Id: number;
  EmployeeId: number;
  TimeOffType: number;
  InitialDateTime: Date;
  EndDateTime: Date;
  Justification: string;
  InitialCompensationDateTime: Date;
  EndCompensationDateTime: Date;
  RequestDate: Date;
  Active: boolean ;
  Employee?: Employee;

  constructor(params?: Partial<TimeOff>) {
    this.Id = params?.Id ;
    this.EmployeeId = params?.EmployeeId ;
    this.TimeOffType = params?.TimeOffType;
    this.InitialDateTime = params?.InitialDateTime;
    this.EndDateTime = params?.EndDateTime ;
    this.Justification = params?.Justification ;
    this.InitialCompensationDateTime = params?.InitialCompensationDateTime;
    this.EndCompensationDateTime = params?.EndCompensationDateTime;
    this.RequestDate = params?.RequestDate;
    this.Active = params?.Active ;
    this.Employee = params?.Employee;

  }
}

File: Api/Timesheet/Entities/TimeOff.cs
namespace Timesheet.Entities
{
    public class TimeOff : EntityBase
    {
        public int TimeOffType { get; set; }
        public DateTime InitialDateTime { get; set; }
        public DateTime EndDateTime { get; set; }
        public string Justification { get; set; }
        public DateTime InitialCompensationDateTime { get; set; }
        public DateTime EndCompensationDateTime { get; set; }
        public DateTime RequestDate { get; set; }

        public bool Active { get; set; }  

        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
    }

}
